src/buildercore/terraform.py

```diff
@@ -697,13 +936,17 @@ def init(stackname, context):
         # TODO: possibly remove unused providers
         # Terraform already prunes them when running, but would
         # simplify the .cfn/terraform/$stackname/ files
-        fp.write(json.dumps({
+        providers = {
             'provider': {
                 'fastly': {
                     # exact version constraint
                     'version': "= %s" % PROVIDER_FASTLY_VERSION,
                     'api_key': "${data.%s.%s.data[\"api_key\"]}" % (DATA_TYPE_VAULT_GENERIC_SECRET, DATA_NAME_VAULT_FASTLY_API_KEY),
                 },
+                'aws': {
+                    'version': "= %s" % '2.3.0',
```

Comment:
minor: not sure about the purpose of the substitution. Are you planning to extract the version into a variable / configuration?

Reply:
separating the `=`, `~=`, etc. modifiers from the version number, ideally the first should all be `=`. It could be configured but no current plans for it.